arduino-ide-extension/src/node/clang-formatter.ts

```diff
@@ -161,3 +137,26 @@ function styleJson({
     UseTab,
   };
 }
+
+export type ClangStyleValue =
+  | string
+  | number
+  | boolean
+  | ClangStyleValue[]
+  | { [key: string]: ClangStyleValue };
+export type ClangConfiguration = Record<string, ClangStyleValue>;
+
+export function stringifyConfiguration(
+  value: ClangStyleValue | ClangConfiguration
+): string {
+  if (Array.isArray(value)) {
+    return `[${value.map((item) => stringifyConfiguration(item)).join(', ')}]`;
+  } else if (typeof value === 'object') {
+    return `{${Object.entries(value)
+      .map(([key, v]) => `${key}: ${stringifyConfiguration(v)}`)
+      .join(', ')}}`;
+  } else if (typeof value === 'string') {
+    return `'${value}'`;
+  }
+  return String(value);
+}
```

Comment:
What is the reason for writing bespoke stringification code instead of using [`JSON.stringify`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify)?

YAML is a superset of JSON, so this is an acceptable data format as far as ClangFormat is concerned.

Reply:
Thanks for asking. I also had difficulty understanding how it should work, and the formatter fix has been implemented via trial and error with tests.

My understanding is that the arguments cannot contain double quotes (`"`) anymore as the process is invoked without a shell, so the argument escaping must be precisely how it is [defined by the formatter](https://clang.llvm.org/docs/ClangFormat.html) (it's for version `17.0`, but the syntax is the same as for `14.0`):

>  4. "{key: value, ...}" to set specific parameters, e.g.:
>      --style="{BasedOnStyle: llvm, IndentWidth: 8}"

As you can see, the `style` "JSON" properties are not escaped with double quotes (`"`). However, IDE2 must escape the `string` values with single quotes (`'`) to support the regexes, such as `'^\"(llvm|llvm-c|clang|clang-c)/'`.

Reply:
Thank you for the help! a790379 includes your suggested changes. I resolve this thread.